x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/Aggregations.java

```diff
@@ -25,7 +25,11 @@ private Aggregations() {}
      */
     enum AggregationType {
         AVG("avg", "double"),
-        MAX("max", null);
+        CARDINALITY("cardinality", "long"),
+        VALUE_COUNT("value_count", "long"),
+        MAX("max", null),
```

Comment:
Shouldn't those 3 get mapped as `double`?

Reply:
both are counts and therefore not floating point types

Reply:
I meant about `max`, `min` & `sum`.

Reply:
`max`, `min` and `sum` are mapped according to the mapping in the source, not visible in the diff as this hasn't been changed, if you expand the diff you will see:

`targetMapping - the field type for the output, if null, the source type should be used`

With other word `null` is a magic value, telling map deduction to use the source mapping.

Reply:
Ah! Cool, thanks for clarifying :-)